js/ElementRendererProvider.js

```diff
@@ -528,6 +550,98 @@ export default class ElementRendererProvider {
       return divCarouselWrapper;
     });
 
+    this.set(TYPES.CAROUSELSELECT, (config): HTMLElement => {
+      const defaultPadding = 0;
+      const padding = config.padding || defaultPadding;
+
+      const carouselWrapper = document.createElement('div');
+      const carousel = document.createElement('div');
+
+      if (config.accessibility && config.accessibility.web) {
+        Utils.appendAttributesFromObject(carouselWrapper, config.accessibility.web);
+      }
+
+      function findCardParent(element: any): HTMLDivElement | typeof undefined {
+        if (!element || element.tagName === 'BUTTON') {
+          return undefined;
+        }
+
+        const index = element.getAttribute('data-carousel-index');
+        if (index !== null) {
+          return element;
+        }
+
+        return findCardParent(element.parentNode);
+      }
+
+      function toggleCardSelect(element: HTMLDivElement, selected: boolean) {
+        if (selected) {
+          element.setAttribute('data-selected', 'true');
+          element.classList.add('lp-json-pollock-layout-selected');
+        } else {
+          element.removeAttribute('data-selected');
+          element.classList.remove('lp-json-pollock-layout-selected');
+        }
+      }
+
+      /**
+       * @param {MouseEvent} event
+       * */
+      function cardClick(event: MouseEvent) {
+        const element = event.target;
+        const cardParent = findCardParent(element);
+
+        if (cardParent) {
+          if (config.selectMode.type === 'single' && cardParent.parentNode) {
+            Array
+              .from((cardParent.parentNode: any).querySelectorAll('[data-carousel-index][data-selected]'))
+              .filter((carouselElement: HTMLDivElement) => carouselElement !== cardParent)
+              .forEach((carouselElement: HTMLDivElement) => {
+                toggleCardSelect(carouselElement, false);
+              });
+          }
+
+          toggleCardSelect(cardParent, cardParent.dataset.selected !== 'true');
+        }
+      }
+
+      /**
+       * Render logic
+       * */
+      (carouselWrapper: any).afterRender = () => {
+        const carouselItemsCount = carouselWrapper.childNodes.length;
```

Comment:
@dvakulenko Hey, not sure if we should use `childNodes`. Along with HTML nodes this can contain text nodes. IMO we need to use `children` instead

Reply:
It's a valid point! Thank you, I removed and used `children` instead.